frontend/src/scenes/surveys/SurveyView.tsx

```diff
@@ -694,7 +697,33 @@ function SurveyNPSResults({
                                     properties: [
                                         {
                                             type: PropertyFilterType.Event,
-                                            key: getSurveyResponseKey(questionIndex),
+                                            key: getResponseFieldWithId(questionIndex, questionId).idBasedKey,
```

Comment:
can we extract those objects to a method and return based on if its `idBasedKey` or `indexBasedKey`? we are duplicating pretty much everything because of the return of `getResponseFieldWithId`